### DISCUSSION TITLE
How to add ui.chip() elements inside a ui.textarea element Without affecting its editable width?

### DISCUSSION THREAD

**Discussion by happybeginning1** [zauberzeug/nicegui#3208] [Q&A] [answered]
### Question

Hello,

I have the following working code:

```python
with ui.textarea(
    value=current_text,
    on_change=lambda e: update_info(e.value, info)
).style('background-color: #3e6a94; width:555px;').props('input-style="color: white" input-class="h-52" outlined').props('bottom-slots') as textarea_note:
    pass
```

Now, I want to add three ui.chip() elements inside the teatarea element, either at the top or at the bottom of this textarea, without occupying the space to its left or right. I attempted the following modification:

```python
with ui.textarea(
    value=current_text,
    on_change=lambda e: update_info(e.value, info)
).style('background-color: #3e6a94; width:555px;').props('input-style="color: white" input-class="h-52" outlined').props('bottom-slots') as textarea_note:
    ui.chip('Text1', selectable=True, color=None)
    ui.chip('Text2', selectable=True, color=None)
    ui.chip('Text3', selectable=True, color=None)
```

However, this results in the ui.chip elements occupying the space to the right of the textarea.

Could someone please guide me on how to achieve the desired effect? Thanks in advance!

**Comment by falkoschindler** [comment, marked as answer]
Hi @happybeginning1,

You can position a row of chips absolutely inside the `ui.input` element:
```py
with ui.textarea().props('outlined').classes('w-96'):
    with ui.row().classes('absolute top-0 right-0 gap-0'):
        ui.chip('Text1')
        ui.chip('Text2')
        ui.chip('Text3')
```

Otherwise the HTML `<input>` element and the three chips would be siblings, sharing the available space.

![Screenshot 2024-06-12 at 15 28 16](https://github.com/zauberzeug/nicegui/assets/5767091/fd7bcc47-5fca-49fb-9041-2ddd3f0f8bd0)

**Reply by happybeginning1** [reply]
Hi @falkoschindler,
It works perfectly. Thank you agin for the help!
